lib/api/utilities/functions.php

```diff
@@ -81,172 +81,173 @@ function beans_remove_dir( $dir_path ) {
 	unset( $items[0], $items[1] );
 
 	foreach ( $items as $needle => $item ) {
-
 		$path = $dir_path . '/' . $item;
 
-		if ( 'dir' === filetype( $dir_path . '/' . $item ) ) {
+		if ( is_dir( $path ) ) {
 			beans_remove_dir( $path );
 		} else {
 			@unlink( $path );
 		}
-
-		unset( $items[ $needle ] );
-
 	}
 
-	@rmdir( $dir_path );
-
-	return true;
-
+	return @rmdir( $dir_path );
 }
 
 /**
  * Convert internal path to a url.
  *
  * This function must only be used with internal paths.
  *
- * @since 1.0.0
+ * @since 1.5.0
  *
- * @param string $path Path to be converted. Accepts absolute and relative internal paths.
+ * @param string $path          Path to be converted. Accepts absolute and relative internal paths.
+ * @param bool   $force_rebuild (Optional) Forces the rebuild of the root url and path.
  *
  * @return string Url.
  */
-function beans_path_to_url( $path ) {
-
-	static $root, $host;
+function beans_path_to_url( $path, $force_rebuild = false ) {
+	static $root_path, $root_url;
 
 	// Stop here if it is already a url or data format.
-	if ( true == preg_match( '#^(http|https|\/\/|data)#', $path ) ) {
+	if ( preg_match( '#^(http|https|\/\/|data)#', $path ) ) {
 		return $path;
 	}
 
 	// Standardize backslashes.
 	$path = wp_normalize_path( $path );
 
 	// Set root and host if it isn't cached.
-	if ( ! $root ) {
+	if ( ! $root_path || true === $force_rebuild ) {
 
 		// Standardize backslashes set host.
-		$root = wp_normalize_path( untrailingslashit( ABSPATH ) );
-		$host = untrailingslashit( site_url() );
+		$root_path = wp_normalize_path( untrailingslashit( ABSPATH ) );
+		$root_url  = untrailingslashit( site_url() );
 
 		// Remove subfolder if necessary.
-		if ( '' !== ( $subfolder = parse_url( $host, PHP_URL_PATH ) ) ) {
-
-			$root = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $root );
-			$host = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $host );
+		$subfolder = parse_url( $root_url, PHP_URL_PATH );
+		if ( $subfolder && '/' !== $subfolder ) {
+			$pattern   = '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#';
+			$root_path = preg_replace( $pattern, '', $root_path );
+			$root_url  = preg_replace( $pattern, '', $root_url );
+		}
 
-			// Add the blog path for multsites.
-			if ( ! is_main_site() && ( $blogdetails = get_blog_details( get_current_blog_id() ) ) ) {
-				if ( ! ( defined( 'WP_SITEURL' ) ) || ( defined( 'WP_SITEURL' ) && WP_SITEURL == site_url() ) ) {
-					$host = untrailingslashit( $host ) . $blogdetails->path;
-				}
+		// If it's a multisite and not the main site, then add the site's path.
+		if ( ! is_main_site() ) {
```

Comment:
Doing the multisite work is not a requirement of the subfolder task.  They are 2 separate tasks.  So I unwrapped it from the subfolder removal section.